### PR TITLE
Don't ignore options diff in synthetics resource

### LINKED ISSUE #697: Changed synthetics_test options to options_list, but options are still in the result of plan.
### Terraform Version

```
Terraform v0.13.0
+ provider registry.terraform.io/datadog/datadog v2.13.0
```

### Affected Resource(s)

- datadog_synthetics_test

### Terraform Configuration Files

This configuration has been changed from the one for terraform-provider-datadog 2.12.x. Changed deprecated `options` to `options_list`.

```hcl
resource "datadog_synthetics_test" "api_test" {
  type    = "api"
  subtype = "http"

  request = {
    method = "GET"
    url    = "https://example.com"
  }

  assertion {
    type     = "statusCode"
    operator = "is"
    target   = 200
  }
  assertion {
    type     = "header"
    property = "content-type"
    operator = "is"
    target   = "application/json"
  }
  assertion {
    type     = "responseTime"
    operator = "lessThan"
    target   = 1000
  }

  locations = [
     "aws:ap-northeast-1",
  ]

  options_list {
    tick_every           = 300
    min_failure_duration = 180
    min_location_failed  = 1
  }

  name    = "API test for example.com"
  message = "API test"
  status = "live"
}
```

### Debug Output

N/A

### Panic Output

N/A

### Expected Behavior

The expected result of the terraform plan is that `options` are removed and `options_list` is added.

```
  # datadog_synthetics_test.api_test will be updated in-place
  ~ resource "datadog_synthetics_test" "api_test" {
    ...
      - options         = {
          - "min_failure_duration" = "180"
          - "min_location_failed"  = "1"
          - "tick_every"           = "300"
        }
      + options_list {
          + min_failure_duration = 180
          + min_location_failed  = 1
          + tick_every           = 300
        }
    }
```

### Actual Behavior

`options` are still there.

```
  # datadog_synthetics_test.api_test will be updated in-place
  ~ resource "datadog_synthetics_test" "api_test" {
    ...
        options         = {
            "min_failure_duration" = "180"
            "min_location_failed"  = "1"
            "tick_every"           = "300"
        }
      + options_list {
          + min_failure_duration = 180
          + min_location_failed  = 1
          + tick_every           = 300
        }
    }
```

### Steps to Reproduce

1. First, in 2.12.x, apply the config with the `options`.
1. Change the `options` to `options_list`.
1.  `terraform plan`

### Important Factoids

N/A

### References

N/A

### PR DESCRIPTION
**Pull request by therve**
We ignore the diff completely when options_list is set, which sounds
like unwanted behavior.

Closes #697

### CODE REVIEW

**Review by romainberger — APPROVED** [review]

**Review by jirikuncar — COMMENTED** [review]
@therve can we add a regression test case?

**Comment by therve** [comment]
> @therve can we add a regression test case?

It's going to be tough because we ignore diffs on those particular fields in the tests. I'll have a look.

File: datadog/resource_datadog_synthetics_test_test.go [hunk 1]
```diff
@@ -263,6 +263,23 @@ func TestAccDatadogSyntheticsTCPTest_Basic(t *testing.T) {
 	})
 }
 
+func TestAccDatadogSyntheticsTCPLegacyOptionsTest_Basic(t *testing.T) {
```
Comment (nmuesch) [inline datadog/resource_datadog_synthetics_test_test.go:266]: Does this test give us anything specific? 
The other two look to cover the case of 
* ensuring we can still set the old options 
* ensuring that if you upgrade from old -> new, the old is no longer present

**Review by nmuesch — APPROVED** [review]
LGTM, left one question about a test

File: datadog/resource_datadog_synthetics_test_test.go (hunk 1 above)
Comment (therve) [reply datadog/resource_datadog_synthetics_test_test.go]: Good point, updated a test instead. We do cover the 2 point.

**Review by nmuesch — APPROVED** [review]

**Pull request merged**